src/cargo/ops/cargo_update.rs

```diff
@@ -207,6 +221,215 @@ pub fn print_lockfile_changes(
+                                        name,
+                                        current,
+                                        latest
+                                    );
+
+                                    opts.gctx.shell().status_with_color(
+                                        "Upgrading",
+                                        format!(
+                                            "{} {} -> v{}",
+                                            name,
+                                            current,
+                                            latest.to_string()
+                                        ),
+                                        &style::GOOD,
+                                    )?;
+
+                                    upgrades.insert(
+                                        name.to_string(),
+                                        latest.clone(),
+                                    );
+
+                                    let req =
+                                        OptVersionReq::Req(VersionReq::parse(&latest.to_string())?);
+                                    let mut dep = dependency.clone();
+                                    dep.set_version_req(req);
+                                    return Ok(dep);
+                                } else {
+                                    trace!(
+                                        "skipping dependency `{}` without a breaking update available",
+                                        name
+                                    );
+                                }
+                                            } else {
+                                trace!(
+                                    "skipping dependency `{}` without any published versions",
+                                    name
+                                );
+                            }
+                        } else {
+                            trace!("skipping non-caret dependency `{}`: {}", name, comparator);
+                        }
+                    } else {
+                        trace!(
+                            "skipping dependency `{}` with multiple version comparators: {:?}",
+                            name,
+                            &current.comparators
+                        );
+                    }
+                } else {
+                    trace!(
+                        "skipping dependency `{}` without a simple version requirement: {}",
+                        name,
+                        version_req
+                    );
+                }
+
+                Ok(dependency)
+            },
+        )?;
+    }
+
+    Ok(upgrades)
+}
+
+/// Update manifests with upgraded versions, and write to disk. Based on cargo-edit.
+pub fn write_manifests(
+    ws: &Workspace<'_>,
+    opts: &UpdateOptions<'_>,
+    upgrades: &HashMap<String, Version>,
+) -> CargoResult<()> {
+    if !opts.breaking {
+        return Ok(());
+    }
```

Comment:
nit: Could we instead do this based on `!upgrades.is_empty()`.  Either way, it should still work.  The benefit comes when we do the other half of this feature which is `--precise <breaking-version>`.

Reply:
Fixed.